src/spin_mutex.cc

```diff
@@ -3,19 +3,40 @@
 #include <cstddef>
 #include <thread>
 
+#define _ASM_NOOP asm volatile("nop")
+
+// Cannot use `do { } while (0)` here because there would be huge differences between -O0 and -O3
+#define _2X(statement) \
+    {                  \
+        statement;     \
+        statement;     \
+    }
+
+#define _2X_NOOP _2X(_ASM_NOOP)
```

Comment:
nop?

Reply:
nop, noop and no-op are the same thing...